### DISCUSSION TITLE
Questions about output of Ambiguity

### DISCUSSION THREAD

**Discussion by LNTU-YS** [groops-devs/groops#292] [Q&A]
Hello,

I have referred to [#135](https://github.com/groops-devs/groops/discussions/135), but I still have some questions to ask you. I hope to receive your help. My settings and [ambiguities_2023-02-02.txt](https://github.com/user-attachments/files/17413720/ambiguities_2023-02-02.txt) are shown below.
![image](https://github.com/user-attachments/assets/c427ee16-606c-4acd-8c2d-13fa6fe540cd)

Questions 1: In the last column of the file I provided, there is a positive number and 915 negative numbers. Does this mean that 915 ambiguities have been fixed?

Questions 2: Does the negative number in the last column represent the residual of ambiguity?

Questions 3:  Why are there some negative numbers in the last column that are less than -0.2, and even less than -0.5, when I set sigmaMaxResolve to 0.2 cycle? Because in my understanding, negative numbers represent fixed , but their absolute values are greater than 0.2, which confuses me.

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Comment by tmayerguerr** [comment]
Hi YUAN,

- Questions 1: correct, 915 ambiguities have been fixed
- Questions 2: exactly, the negative number represent the residual of the fixed ambiguity
- Questions 3: The sigmaMaxResolve is related to the formal accuracy (variance of the estimated parameter of the least squares adjustment). The actual estimated value may deviate from this and may also be greater. There are indeed opportunities to improve this in the future.

best regards
Torsten

**Reply by LNTU-YS** [reply]
Hi Torsten, 

Thank you very much for your help. Regarding question 1, I have a new question to ask you for advice. According to my rough understanding, if there are 5.2 and 5.9, after fixing the ambiguity, it should be 5 and 6, which means a change of -0.2 and 0.1. So I think the residual of the fixed ambiguity should have both positive and negative values, but the fact is that they are all positive, and the negative sign is used to distinguish whether they are fixed or not.

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Comment by tmayerguerr** [comment]
Hi YUAN,

For the unfixed float ambiguities, the nearest integer is subtracted and then the absolute value is calculated. For the residuals of the fixed ambiguities, the absolute is calculated and then a minus is written in front of it (to distinguish between float and fixed).

This is computed in gnssLambda.cpp, line 333
```
      for(UInt i=0; i<idxSolved; i++)
        xBar(i) = std::fabs(std::remainder(xBar(i), 1));
      for(UInt i=idxSolved; i<dim; i++)
        xBar(i) = -std::fabs(xBar(i));
```
Hopes this helps.
regards
Torsten

**Reply by LNTU-YS** [reply]
Hi Torsten,

Thank you very much for your help. I have gained a lot from it.

With Kind Regards
YUAN

**Comment by LNTU-YS** [comment]
Hi @tmayerguerr ,

I have a follow-up question for question 3.

I would like to ask if there are any other detection conditions in AR besides setting sigmaMaxResolve to 0.2 cycle, such as the setting of "ratio>2".

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Reply by LNTU-YS** [reply]
Hi,

I noticed Monte Carlo in AR today, and I think this is the answer I was looking for. Am I correct?
